pkg/apis/management.cattle.io/v3/authz_types.go

```diff
@@ -89,6 +89,39 @@ type GlobalRole struct {
 	Rules          []rbacv1.PolicyRule `json:"rules,omitempty"`
 	NewUserDefault bool                `json:"newUserDefault,omitempty" norman:"required"`
 	Builtin        bool                `json:"builtin" norman:"nocreate,noupdate"`
+
+	// InheritedClusterRoles are the names of RoleTemplates whose permissions are granted by this GlobalRole in every
+	// cluster besides the local cluster. To grant permissions in the local cluster, use the Rules or NamespacedRules
+	// fields.
+	// +optional
+	InheritedClusterRoles []string `json:"inheritedClusterRoles,omitempty"`
+
+	// NamespacedRules are the rules granted by this role in one specific namespace of the cluster. Will produce an
+	// error if the namespace specified in the key does not exist.
+	// +optional
+	NamespacedRules map[string][]rbacv1.PolicyRule `json:"namespacedRules,omitempty"`
+
+	// Status tracks the status of the backing RBAC that this GlobalRole is orchestrating
+	// +optional
+	Status *GlobalRoleStatus
+}
+
+const (
+	GlobalRoleStatusSummaryComplete   = "Complete"
+	GlobalRoleStatusSummaryInProgress = "InProgress"
+	GlobalRoleStatusSummaryError      = "Error"
+)
+
+type GlobalRoleStatus struct {
+	// LastUpdate is the last time that this status was updated
+	// TODO: Add format of timestamp
+	LastUpdate string
+
+	// Summary is the summary of the various conditions - one of Complete, InProgress, or Error
+	Summary string
```

Comment:
I think you are right here - this being said, this field isn't really being modified by this PR. I can remove the other new fields outside of the `inheritedClusterRoles` if you would prefer, but I'm not going to make further adjustments to this in the scope of this PR - there's enough changes as-is.

Reply:
But Summary is a new field, right?

Reply:
It's a new field, but only being addressed here: https://github.com/rancher/rancher/issues/42217. I've added it here so that others can get an idea of the bigger picture, but won't address it until I pick up that issue.

Like I said above, I can either remove it or leave as-is - let me know which option you prefer.

Reply:
I can do that, will remove the other fields.